lib/trento/activity_log.ex

```diff
@@ -42,14 +40,51 @@ defmodule Trento.ActivityLog do
     end
   end
 
-  @spec list_activity_log() :: list(ActivityLog.t())
-  def list_activity_log do
-    # This will be made filterable/paginatable in a later PR
-    query =
-      from activity in ActivityLog,
-        order_by: [desc: activity.inserted_at]
+  @spec list_activity_log(map()) ::
+          {:ok, list(ActivityLog.t()), Flop.Meta.t()} | {:error, :activity_log_fetch_error}
+  def list_activity_log(params) do
+    parsed_params =
+      case params do
+        [] -> %{}
+        params -> parse_params(params)
+      end
+
+    case Flop.validate_and_run(ActivityLog, parsed_params, for: ActivityLog) do
+      {:ok, {activity_log_entries, meta}} ->
+        {:ok, activity_log_entries, meta}
+
+      error ->
+        Logger.error("Activity log fetch error: #{inspect(error)}")
+        {:error, :activity_log_fetch_error}
+    end
+  end
+
+  # ''&& false' is a workaround until we reach OTP 27 that allows doc tag for private functions;
+  # we get a compile warning without this with OTP 26
+  @doc """
+       Parses the query parameters and returns a map with the parsed values as expected by the Flop library.
+       Some parameters are recognized by Flop and are used as is (example: last, first, after, before);
+       some other parameters are used to build filters with custom operator logic (example: from_date, to_date, actor, type).
+       ## Examples
+             iex> parse_params([{:from_date, "2021-01-01"}, {:to_date, "2021-01-31"}, last: 10])
+             %{filters: %{inserted_at: %{value: "2021-01-31", op: :>=, field: :inserted_at}}, last: 10}
+       """ && false
+  defp parse_params(query_params) do
+    query_params
+    |> Enum.map(fn
+      {:from_date, v} -> {:filters, %{field: :inserted_at, op: :<=, value: v}}
+      {:to_date, v} -> {:filters, %{field: :inserted_at, op: :>=, value: v}}
+      {:actor, v} -> {:filters, %{field: :actor, op: :ilike_or, value: v}}
+      {:type, v} -> {:filters, %{field: :type, op: :ilike_or, value: v}}
+      param -> param
+    end)
+    |> Enum.reduce(%{filters: %{}}, fn
+      {:filters, filter}, acc ->
+        Map.put(acc, :filters, Map.put(acc.filters, filter.field, filter))
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** This introduced a bug. Using filters as a map only allows for one filter per each field. If we have multiple filters for the same field (like dates for examples) they won't be considered. (my bad, the refactor was my suggestion).

Using filters like collection it would solve the issue.

```suggestion
    |> Enum.reduce(%{filters: []}, fn
      {:filters, filter}, acc ->
        Map.put(acc, :filters, [filter | acc.filters])
```

However, by changing the shape of filters I expect tests to fail, but they don't

‏